evalml/pipelines/time_series_classifcation_pipelines.py

```diff
@@ -0,0 +1,204 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.classification_pipeline import ClassificationPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    _get_rows_without_nans,
+    drop_rows_with_nans,
+    pad_with_nans
+)
+
+
+class TimeSeriesClassificationPipeline(ClassificationPipeline):
+    """Pipeline base class for time series classifcation problems."""
+
+    def __init__(self, parameters, random_state=0):
+        """Machine learning pipeline for time series classification problems made out of transformers and a classifier.
+
+        Required Class Variables:
+            component_graph (list): List of components in order. Accepts strings or ComponentBase subclasses in the list
+
+        Arguments:
+            parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
+                 An empty dictionary {} implies using all default values for component parameters. Pipeline-level
+                 parameters such as gap and max_delay must be specified with the "pipeline" key. For example:
+                 Pipeline(parameters={"pipeline": {"max_delay": 4, "gap": 2}}).
+            random_state (int, np.random.RandomState): The random seed/state. Defaults to 0.
+        """
+        if "pipeline" not in parameters:
```

Comment:
Really nit-picky, but what if a user passes in a "pipeline" key mapping to a dictionary with "gap" or "max_delay" missing? Ex: {pipeline: {"gap":1}} I think we get a KeyError in the following lines, right?

Reply:
Yep you're right!

Reply:
We'll have to fix the same thing in regression pipelines so I'll just file an issue and change it for both after this is merged!